modules/graphql_image/src/Plugin/GraphQL/Fields/ImageDerivative.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\graphql_image\Plugin\GraphQL\Fields;
+
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
+use Drupal\Core\Image\ImageFactory;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Drupal\graphql_core\GraphQL\FieldPluginBase;
+use Drupal\image\Plugin\Field\FieldType\ImageItem;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Youshido\GraphQL\Execution\ResolveInfo;
+use Drupal\image\Entity\ImageStyle;
+
+/**
+ * Retrieve the image field derivative (image style).
+ *
+ * @GraphQLField(
+ *   id = "image_derivative",
+ *   name = "derivative",
+ *   type = "ImageStyle",
+ *   nullable = true,
+ *   arguments = {
+ *     "style" = "ImageStyleId"
+ *   },
+ *   types = {
+ *     "Image"
+ *   }
+ * )
+ */
+class ImageDerivative extends FieldPluginBase implements ContainerFactoryPluginInterface {
+
+  use DependencySerializationTrait;
+
+  /**
+   * The image factory.
+   *
+   * @var \Drupal\Core\Image\ImageFactory
+   */
+  protected $imageFactory;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $pluginId, $pluginDefinition, ImageFactory $imageFactory) {
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->imageFactory = $imageFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $pluginId, $pluginDefinition) {
+    return new static($configuration, $pluginId, $pluginDefinition, $container->get('image.factory'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function resolveValues($value, array $args, ResolveInfo $info) {
+    if ($value instanceof ImageItem && $style = ImageStyle::load($args['style'])) {
+      $file = $value->entity;
+      $styleUri = $style->buildUri($file->getFileUri());
+      if (!file_exists($styleUri)) {
```

Comment:
Yeah no. Let's not create files during the request. That's the job of the file controller upon an actual request.

Instead, you can pre-calculate / derive the expected width/height like this: 

```


  /**
   * Returns the expected image dimensions.
   *
   * @return array The image dimensions as an associative array.
   */
  protected function calculateDimensions() {
    if (isset($this->dimensions)) {
      return $this->dimensions;
    }

    $this->dimensions = [
      'height' => $this->height,
      'width' => $this->width,
    ];

    $this->style->transformDimensions($this->dimensions, $this->file->getFileUri());

    return $this->dimensions;
  }
```